hal/src/utility.ts

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) Microsoft Corporation. All Rights Reserved. */
+
+import {hal} from './decorators';
+import {Rel, Hal} from './constants';
+import {Server} from './server';
+import {Response} from './response';
+import {Template} from './template';
+
+export function Utility(server: Object): Utility.Server {
+    return {
+        links: ServerUtility.links.bind(server),
+        href: ServerUtility.href.bind(server)
+    };
+}
+
+class ServerUtility {
+    static links(this: Object, rel: Rel, overrides?: hal.Overrides): Hal.Link[] {
```

Comment:
i still don't know how i feel about doing this. and if you're going to go to the trouble of typing it, why is it Object?